jQuery update, remove footnotes, improve permalinks and internal link fix

MightyCreak opened the pull request:
@Xenopathic I've separated everything into 4 distinct commits.

There are some trailing spaces in tipsy, but since it's an external lib, I don't really care :P

Xenopathic commented:
That 'update' of Tipsy, is actually a downgrade. We were using the master copy from https://github.com/CloCkWeRX/tipsy, which is actually maintained, not the original which stopped being maintained and has many bugs.

Xenopathic commented:
With regards to 'Permalink improvements': you are replicating the `hover` selector of CSS, in Javascript. So why not use the `hover` selector?

```
diff --git a/http/css/style.css b/http/css/style.css
index d0864cb..7a6c300 100644
--- a/http/css/style.css
+++ b/http/css/style.css
@@ -65,10 +65,14 @@ p { max-width: 800px; }
 .hCell-sep       { width: 0.3em; }
 .hCell-driver    { min-width: 5.1em; }

-.permalink { margin-left: 0.3em; }
+.permalink { margin-left: 0.3em; display:none; }
 .permalink:link { text-decoration: none; color: #DDDDDD; }
 .permalink:visited { text-decoration: none; color: #DDDDDD; }

+:hover > .permalink {
+    display: initial;
+}
+
 .extension-child { padding-left: 1em; }
 .extension:hover { background-color: #babdb6; }
 .task {
diff --git a/http/js/script.js b/http/js/script.js
index 377f0b5..d947319 100644
--- a/http/js/script.js
+++ b/http/js/script.js
@@ -89,12 +89,4 @@ $(document).ready(function() {
         var timeDiff = (Date.now()/1000) - $(this).data('timestamp');
         $(this).css('opacity', gaussian(timeDiff, 1, 0, timeConst));
     });
-
-    $('.permalink').hide();
-    $(':has(.permalink)').mouseenter(function() {
-        $(this).children('.permalink').show();
-    });
-    $(':has(.permalink)').mouseleave(function() {
-        $(this).children('.permalink').hide();
-    });
 });
```

MightyCreak commented:
For tipsy, it's little bit hard to understand, there are copies everywhere... I downloaded what seemed to be the latest version on NPM (the recommended way to download a jQuery plugin apparently). I'll try with the latest code from CloCkWeRX, but for sure, with the previous version (before this commit), there was a glitch when moving the mouse between the date and the table cell.

For the permalink, I tried to do it with CSS, but I couldn't figure out which selector I should use to select the child inside the element you're hovering. Now that I see your solution, it is crystal clear ;)

MightyCreak commented:
Ok, I just got the latest tipsy code from CloCkWeRX (there are still a lot of changes, but it still fixes the small glitch when hovering the tasks with text in them).

I also removed the javascript code for the permalink and replaced it by you CSS code (much better! :+1: )

This should be much better I hope! :smile:

Xenopathic commented:
Clarification request: the contribution guidelines state that after a `{`, the body of a control block should be on a new line - does this also apply to CSS? Personally I can't stand the `abc { display:none; }` style :speak_no_evil: but I'll use it if you want.

Xenopathic commented on http/css/style.css:
```diff
@@ -65,6 +65,11 @@ p { max-width: 800px; }
+.permalink { margin-left: 0.3em; display: none; }
+.permalink:link { text-decoration: none; color: #DDDDDD; }
+.permalink:visited { text-decoration: none; color: #DDDDDD; }
+:hover > .permalink { display: inline; }
```
If you use `display: initial;` here instead, this will work in the future if/when `.permalink` is a block element, instead of an inline element.

MightyCreak replied:
I tried with that first, and it also worked, but I haven't found any reference to `initial` value in the standard: https://developer.mozilla.org/fr/docs/Web/CSS/display, so I'm not very keen to use it :neutral_face:

Xenopathic commented on http/css/style.css:
```diff
@@ -74,6 +79,7 @@ p { max-width: 800px; }
     height: 19px;
     width: 72px;
     padding: 0;
+    cursor: default;
```
Where do these `cursor: default`s keep creeping in? It's implied as standard...

Xenopathic replied:
Nevermind, I just read your reply to the previous commit (which disappeared from this PR so I didn't see it). Makes sense now.

MightyCreak replied:
:smile:

Xenopathic commented:
Looks like the Tipsy code still isn't the latest from CloCkWeRX...

MightyCreak commented:
I actually have no coding standard for CSS, but I suppose adapting the one we already have for the other languages is the right way to go.

I'll recheck for Tipsy (maybe I messed up with the rebase...)

MightyCreak commented:
I updated to the very last code from Clockwerx (he should update his README file, I copy/pasted the git url from it).

Now the problem is that Tipsy just doesn't work anymore...

MightyCreak commented:
I think the last version of the Clockwerx repository just doesn't work. I'll revert it and only keep the jQuery update.

MightyCreak commented:
Here it is, everything but the tipsy update. The only problem is that there is a glitch on the footnote displaying when switching between the task and the date inside the task.

Do you validate the PR?

Xenopathic commented on http/js/script.js:
```diff
@@ -80,19 +80,12 @@ function gaussian(x, a, b, c) {
-//        }
-//    });
+    // Add tipsy for the footnote.
+    $('.footnote').tipsy({title: 'data-footnote', gravity: 'w', fade: true});
```
Out of interest, what was the rationale behind changing from `title` to `data-footnote`? The `title` attribute is suited to this purpose, of displaying a tooltip on an element. It also works if JS is disabled (albeit without the fancy Tipsy tooltip)

MightyCreak replied:
You've got a point here!

Actually, I was so used to set titles only for `<a>` tags that I forgot that it could be set on every tags. I'll fix that.

MightyCreak replied:
Fixed!

Xenopathic commented:
Nice :+1:
